### ISSUE TITLE
S3260 triggers on static classes

### ISSUE THREAD

**Issue by rsking** [SonarSource/sonar-dotnet#4980]
### Description

The rule to seal private classes that are not derived (S3260), also triggers on `static` classes, that cannot be `sealed`.

### Repro steps

``` csharp
public class Class1
{
    public string Value1 => Class2.Value2;

    private static class Class2
    {
        public const string Value2 = nameof(Value2);
    }
}
```

### Expected behavior

No warning from S3260, as `class` is `static`.

### Actual behavior

Warning produced for all `private static` classes.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* `SonarAnalyzer.CSharp` version `8.30.0.37606`
* VS2022 RC (Version 17.0.0 RC) / VS2019 (16.11.5)
* dotnet SDK 6.0.100-rc.2.21505.57 / dotnet SDK 5.0.402
* Windows 10 (19043.1266)

**Comment by mary-georgiou-sonarsource** [comment]
Hello, and thank you for reporting these FPs. 
This issue has been fixed here https://github.com/SonarSource/sonar-dotnet/pull/4958, and the fix will be included in the next release.
